moveit_experimental/moveit_jog_arm/src/collision_check_thread.cpp

```diff
@@ -98,6 +102,10 @@ void CollisionCheckThread::startMainLoop(JogArmShared& shared_variables)
       // Do a thread-safe distance-based collision detection
       getLockedPlanningSceneRO()->checkCollision(collision_request, collision_result, current_state);
```

Comment:
Looking at the implementation of [checkCollision()](https://github.com/ros-planning/moveit/blob/master/moveit_core/planning_scene/src/planning_scene.cpp#L482) we are still doing one redundant distance check with the scene. We could remove one check by running `getCollisionEnv()->checkRobotCollision()` and `getCollisionEnvUnpadded()->checkSelfCollision()` separately and then checking the distance and collision results for both.

Reply:
I tried this, but it's always returning 1.797e308 (i.e. DOUBLE_MAX) for the scene collisions, and 0. or -1. for the self-collisions. I'd appreciate a quick sanity check, if you could give it a look.

Reply:
Well that's odd. I don't see anything wrong with the code, though. Do you have collision objects in the planning scene (not part of the urdf)? I would expect it to return DOUBLE_MAX in case there are no objects to collide with. For self collisions we should look up how the values are computed. Feel free to start digging in FCL and ping me if you find something, I'll try to debug it later as well.

Reply:
I ran it in some more scenarios (two other setups in our current project), and the scene collision distances look believable: about 0.1-0.2m. I'm looking into the self-collision.

Reply:
Adding the ACM to the `checkSelfCollision` call fixed it.